Backend/utils/MentiiAuth.py

```diff
@@ -5,90 +5,83 @@
 import ConfigParser as cp
 from mentii import user_ctrl
 
-class MentiiAuthentication:
-
-  def __init__(self):
-    return
-
-  @staticmethod
-  def build_user_object(userData):
-    '''
-    Builds a standard user object to return to the client for all authenticated calls
-    '''
+def build_user_object(userData):
+  '''
+  Builds a standard user object to return to the client for all authenticated calls
+  '''
+  if 'email' in userData:
     return {'email': userData['email']}
+  else:
+    # Anything other than raising an exception here would be a security flaw.
+    # If the code executes to this point and userData does not contain an email
+    # then something is very bad. If only "None" was returned then the faulty
+    # user would still be given access to resources that they shouldnt be allowed
+    raise ValueError('This should not be called anywhere other than within MentiiAuthentication. If email is not an attribute of userData then somehow the user has managed to bypass authentication without an email and should be stopped.')
 
-  @staticmethod
-  def get_app_secret():
-    '''
-    Reads from the prodConfig.ini file for the appSecret
-    '''
-    parser = cp.ConfigParser()
-    parser.read("/config/prodConfig.ini")
-    return parser.get('MentiiAuthentication', 'appSecret')
-
-  @staticmethod
-  def generate_auth_token(userCredentials, expiration = 86400):
-    '''
-    Generates the auth token based off of the user's email and password with a default expiration duration.
-    Default token will last 1 day.
-    '''
-    retval = None
-    try:
-      appSecret = MentiiAuthentication.get_app_secret()
-      s = Serializer(appSecret, expires_in = expiration)
+def generate_auth_token(userCredentials, appSecret=None, expiration = 86400):
+  '''
+  Generates the auth token based off of the user's email and password with a default expiration duration.
+  Default token will last 1 day.
+  '''
+  if appSecret == None:
+    raise ValueError('appSecret cannot be none')
+  retval = None
+  try:
+    s = Serializer(appSecret, expires_in = expiration)
+    if 'email' in userCredentials and 'password' in userCredentials:
       email = userCredentials['email']
       password = userCredentials['password']
       retval = s.dumps({ 'email': email, 'password': password })
-    except:
-      retval = None
-    return retval
-
-  @staticmethod
-  def verify_auth_token(token):
-    '''
-    Reads the auth token to make sure it has not been modified and that it is not expired
-    '''
+  except:
     retval = None
-    appSecret = MentiiAuthentication.get_app_secret()
-    s = Serializer(appSecret)
-    try:
-      user = s.loads(token)
-      retval = user # user from token
-    except SignatureExpired:
-      retval = None # valid token, but expired
-    except BadSignature:
-      retval = None # invalid token
-    return retval
+  return retval
+
+def verify_auth_token(token, appSecret=None):
```

Comment:
Just a quick way to explicitly set it to none and do error checking on it if it's not provided